Rewrite create-torrent script

zpc0 opened the pull request:
This is Draft.
Current create-torrent problems:
- On Debian, creating v2 torrent isn't supported.(And mktorrent is outdated software)
- Setting custom webseed hosts isn't supported.

Creating Python script which uses libtorrent-rasterbar Python bindings will resolve both.
- Both Debian(12+) and Fedora provides libtorrent-rasterbar python bindings package
- Setting custom properties is easier

This is draft because supporting custom webseed url is in progress.
Any reviews are welcome!

marmarek committed:
template: create filesystem on EFI partition

Bootloader isn't installed there yet, but having the filesystem allows
systemd to mount it (it really wants to, even if not booing in EFI
mode...) instead of failing.
And also, set 4k sector size to avoid compatibility issues.

QubesOS/qubes-issues#4974
Fixes QubesOS/qubes-issues#8954

DemiMarie committed:
Use hardened DNF configuration in mock

deltarpm and zchunk are unwanted attack surface.

zpc0 committed:
installer: rewrite create-torrent

- Python based
- Can create v2 hybrid torrent on both Debian and Fedora
- Avoid use of outdated mktorrent
- python3-libtorrent(Debian) or rb_libtorrent-python3(Fedora) are required
- To set custom WebSeed URL, place URL under
  "repository_upload_remote_host" config with "webseed" id.

marmarek commented:
You can drop the .py extension, it isn't needed for scripts (only modules need it). See also formatting issues flagged by CI.

zpc0 commented:
> You can drop the .py extension, it isn't needed for scripts (only modules need it). See also formatting issues flagged by CI.

Thank you for review. Lint issues are now fixed, and I removed .py extension.

marmarek committed:
Add dist-upgrade component

QubesOS/qubes-issues#9317

marmarek committed:
example-configs: add xfwm4 4.20 to R4.2 config for F42 template

QubesOS/qubes-issues#9807

dill-shower committed:
Fix syntax error

omeg committed:
windows: fix build failure detection for ssh executor

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>

Commit:
fix(arch): create devtools makepkg.conf.d/

Commit:
fix(arch): create chroot pacman.conf.d in generate-pacman

Commit:
fix: qualify dockerfile base image prefixes

Commit:
example-configs/archlinux: remove non-existing pkgs bin86,dev86

marmarek committed:
Merge remote-tracking branch 'origin/pr/204'

* origin/pr/204:
  example-configs/archlinux: remove non-existing pkgs bin86,dev86

Pull request description:

`init-cache` fails for archlinux example due to missing packages `bin86`,`dev86`. Fix.

marmarek committed:
Merge remote-tracking branch 'origin/pr/176'

* origin/pr/176:
  Use hardened DNF configuration in mock

marmarek committed:
Merge remote-tracking branch 'origin/pr/203'

* origin/pr/203:
  fix(arch): create chroot pacman.conf.d in generate-pacman
  fix(arch): create devtools makepkg.conf.d/

Pull request description:

Create local `devtools` configuration directories before writing to them.

marmarek committed:
example-configs: remove outdated packages from R4.3

The Fedora 41 used as dom0 base has newer packages already. Few comments
about patched packages:
- libwnck3 - the old version had Fedora downstream patch that used cairo
  to render icons, which was buggy
  (https://gitlab.gnome.org/GNOME/libwnck/-/merge_requests/10#note_2176858,
   QSB-104). This patch was dropped upstream, so no need for own package
  anymore
- pykickstart, lorax - there used to be patches to add package
  verification, but now it's handled natively in lorax (via repository
  file, instead of kickstart repo lines)

QubesOS/qubes-issues#9402

marmarek committed:
Include Git in Windows build VM

Easiest method to apply a patch in Windows.

fepitre committed:
Support executor cleanup on interrupt

Closes QubesOS/qubes-issues#9588

marmarek committed:
Make debian-get-packages-list compatible with python-debian 0.1.51

Package-list is a list of proper dicts already, no need to split them
manually. This interface is already part of python-debian 1.0.0 too, so
there is a chance it won't change drastically anymore.

marmarek committed:
example-configs: update default distributions for VMs

Switch to Fedora 42 and Debian bookworm in VM.
Keep bookworm too, as it's used by Whonix 17.

QubesOS/qubes-issues#8841
QubesOS/qubes-issues#9807

zpc0 committed:
create-torrent: replace dead trackers

Replaces dead HTTP and HTTPS trackers to UDP.

marmarek committed:
Update base builder container to Fedora 42

QubesOS/qubes-issues#9807

marmarek committed:
Install 'hostname' in build container

pbuilder needs it.

marmarek committed:
Add Fedora 43 key

QubesOS/qubes-issues#10102

marmarek committed:
R4.3: add ansible

QubesOS/qubes-issues#10004

fepitre committed:
container: set container in the instance

omeg committed:
windows: Fix local build

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>

fepitre committed:
container: fix error message reported on copy in/out

fepitre committed:
container: ensure self.container exists

fepitre committed:
qubes: create common cleanup

fepitre committed:
cli_base: print traceback on cleanup

marmarek committed:
Merge remote-tracking branch 'origin/pr/206'

* origin/pr/206:
  cli_base: print traceback on cleanup
  qubes: create common cleanup
  container: ensure self.container exists
  container: fix error message reported on copy in/out
  container: set container in the instance
  Support executor cleanup on interrupt

Pull request description:

Closes QubesOS/qubes-issues#9588

fepitre committed:
Make common save_artifacts_info function for plugins

fepitre committed:
Do package init-cache automatically

Fix also few implicit returns

Closes QubesOS/qubes-issues#9515

marmarek committed:
ci: include fc43 cache job

QubesOS/qubes-issues#10102

alimirjamali committed:
Initialize asyncio event loop before using it

Python 3.14 (in Fedora 43) throws RunetimeError if event loop is
not initialized before using it.

resolves: https://github.com/QubesOS/qubes-issues/issues/10188

marmarek committed:
Quick band-aid for builderv2-github compat

builderv2-github doesn't use click, but it does call `_*_stage` functions.
Apply quick fix for now, but it needs a proper solution.

QubesOS/qubes-issues#10233

omeg committed:
windows: fix rpc service copying

By default QWT is installed in Program Files which is not writable by standard users.
Use the override RPC location in the private volume instead.

marmarek committed:
example-configs: backport xfce4-screensaver update for R4.3

marmarek committed:
example-configs: update example templates to F42 and D13 too

Earlier commit updated default distributions, but didn't update template
list.

Fixes: ca2d25c "example-configs: update default distributions for VMs"
Fixes QubesOS/qubes-issues#10252

marmarek committed:
example-configs: switch Whonix 17 branch to bookworm

The qubes-template-whonix repository has separate branches for Whonix
17 and Whonix 18 now - bookworm and trixie. Switch R4.2 (that has Whonix
17) to bookworm one.

QubesOS/qubes-issues#10253

marmarek committed:
example-configs: Set configs for Whonix 18 in R4.3

QubesOS/qubes-issues#10253

marmarek committed:
ci: switch to Whonix 18 and Kicksecure 18

Due to a single template-whonix/template-kicksecure component that has
now separate branches for version 17 and 18, it's not possible to have
both versions in the same config. Switch to the newer version in CI.

QubesOS/qubes-issues#10253

marmarek committed:
example-configs: update kicksecure config to trixie too

QubesOS/qubes-issues#10253

marmarek committed:
tests: update whonix reference

QubesOS/qubes-issues#10253

marmarek committed:
tests: bypass signed tag requirement on template-kicksecure repo

Ease testing pre-releases. Keep the requirement in example configs.

marmarek committed:
Fix chroot cache path

The chroot cache path was changed from just distribution code name to
full package set (so, "vm-bookworm" instead of just "bookworm"). But
that change missed few spots. Adjust them too.

Fixes: fd81b5c "Do package init-cache automatically"

marmarek committed:
Merge branch 'chroot-dir'

* chroot-dir:
  Fix chroot cache path

marmarek committed:
Merge remote-tracking branch 'origin/pr/215'

* origin/pr/215:
  windows: fix rpc service copying

marmarek committed:
example-configs: add core-admin-addon-kicksecure

Needed for sdwdate to work on Kicksecure 18.

Fixes QubesOS/qubes-issues#10020

fepitre committed:
Improve job resolutions by automatically trigger missing deps

omeg committed:
Remove +x attribute from python file

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>

omeg committed:
windows: properly catch build errors

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>

omeg committed:
windows: minor logging improvements

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>

omeg committed:
windows: don't use redundant directories for build output

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>

omeg committed:
windows: change script line endings to LF

Signed-off-by: Rafał Wojdyła <omeg@invisiblethingslab.com>

fepitre committed:
Do not fetch if not explicitely called from CLI

fepitre committed:
Improve tmpl dependencies and fix ComponentDependency in config

fepitre committed:
Improve handling of fetch stage

skip-git-fetch is set to True by default.
Now fetch is enforced by specifically calling fetch command.
We trigger fetch stage in all the cases in order to ensure
that sources are available. This is mandatory to compute
all the dependencies notably to know if for a given distribution
there will be packages to be built later.

fepitre committed:
Check stage in publish or sign from_args

fepitre committed:
test_log: fix tests

fepitre committed:
Update and fix chroot plugins

fepitre committed:
plugins: provide kwargs in run

marmarek committed:
example-configs: update R4.2 default template to F42

QubesOS/qubes-issues#9807

marmarek committed:
Merge remote-tracking branch 'origin/pr/213'

* origin/pr/213:
  Initialize asyncio event loop before using it

marmarek committed:
example-configs: switch video-companion to release-specific branch

It will use R4.3+ qvm-device API, so don't build the new version on R4.2

QubesOS/qubes-issues#10411

fepitre committed:
Several improvements for jobs resolution and cache

fepitre committed:
Update mock dockerfile for Fedora42+

fepitre committed:
tests: update test names for fedora minimal

marmarek committed:
example-configs: add forked xorg-x11-server to R4.3 too

https://github.com/Dasharo/dasharo-issues/issues/949

fepitre committed:
Improve template dependency resolution based on timestamp

marmarek committed:
example-configs: switch R4.3 to release4.3 branch

marmarek committed:
Prevent installing template rpms directly

Installing templates using dnf/rpm directly leads to several issues,
prevent doing it replacing all pre/post scripts with a simple
pre-install script that always fails with a message suggesting to use
qvm-template.

Using a non-existing dependency might be better, but it breaks
downloading the package using dnf for the installation image.

Fixes QubesOS/qubes-issues#9943

fepitre committed:
Several improvements for JobDependency and template plugin

fepitre committed:
tests: use 4.3 for template jobs

marmarek committed:
Set Arch's multilib repo to DatabaseOptional

pacman 7.1.0 changed default for SigLevel to Required, which doesn't match
official mirrors (they have signed packages, but not databases).
Normally pacman.conf has `Siglevel = Required DatabaseOptional` for
official repositories, but the one in the builderv2 had just
`PackageRequired`. This used to be fine global default optional, but it
isn't anymore.
'core' and 'extra' already has DatabaseOptional, change it in 'multilib'
too

marmarek committed:
Fix container tests

Looks like in recent docker the in-container root cannot write to
all files anymore. ContainerExecutor accounts for that by adjusting
permissions already, but only in a specific directory used by the
builder.
Make the test use that directory for more realistic test.

marmarek committed:
example-configs: enable vm-fc43 in R4.3 and main

QubesOS/qubes-issues#10102

marmarek committed:
Consistently use config.parse_qubes_release()

Template plugin used both manual regex parse and parse_qubes_release()
function. Switch to the latter consistently.

marmarek committed:
Parse 'devel' version as '99.0'

This (hopefully) guarantee to be always newer than any released version.

QubesOS/qubes-issues#10478

marmarek committed:
Drop argument verification in create-skeleton scripts

Those are internal scripts called by qubes-builderv2 which already
ensures well formed arguments. Do not duplicate verification logic, as
that makes it harder to keep them consistent.

QubesOS/qubes-issues#10478

marmarek committed:
tests: test 'devel' repository paths

Check if non-numeric release is handled correctly.

marmarek committed:
Merge branch 'repo-devel-cont'

* repo-devel-cont:
  tests: test 'devel' repository paths
  Drop argument verification in create-skeleton scripts
  Parse 'devel' version as '99.0'
  Consistently use config.parse_qubes_release()

marmarek committed:
example-configs: set version to 'devel' in qubes-os-main.yml

QubesOS/qubes-issues#10478

marmarek committed:
Preserve explicit skip-git-fetch value

If skip-git-fetch is set explicitly in the config, do not override it
with implicit value based on 'fetch' stage being requested or not.

This fixes cases where source dir is managed externally (like in CI),
but fetch is still needed to download auxiliary sources (tarballs etc).

alimirjamali committed:
Prevent `MAINTAINER instruction is deprecated` error

According to:
https://docs.docker.com/reference/build-checks/maintainer-deprecated/

Also a small improvement to container generator

marmarek committed:
Refresh archlinux keys before building chroot

The keys packaged in distributions are not always the freshest one.
Refresh them from keyservers to get updated subkeys/signatures,
otherwise package verification fails, for example like this:

    [qb.chroot_archlinux.vm-archlinux.init-cache] error: libseccomp: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust
    [qb.chroot_archlinux.vm-archlinux.init-cache] error: failed to commit transaction (invalid or corrupted package)

marmarek committed:
Merge remote-tracking branch 'origin/pr/226'

* origin/pr/226:
  Prevent `MAINTAINER instruction is deprecated` error

marmarek committed:
Do not fail init-cache if some keys cannot be refreshed

Keyservers are not reliable nowadays, especially over tor. Do key
refresh on best effort basis, but don't fail if some cannot be updated.
It's enough to get keys used for signing current packages (used in the
chroot).

Fixes: 7cc4814 "Refresh archlinux keys before building chroot"

marmarek committed:
Fix archlinux cache path

chroot_archlinux plugin saves it in self.dist.nva subdir.

Related to QubesOS/qubes-issues#10528

fepitre committed:
chroot_archlinux: irgnore if an error occured on creating archive

Closes QubesOS/qubes-issues#10528

adrelanos committed:
update Kicksecure, Whonix signing key

same key fingerprint, without expiration date

marmarek committed:
Merge remote-tracking branch 'origin/pr/229'

marmarek committed:
Merge branch 'template-rpm'

* template-rpm:
  Prevent installing template rpms directly

marmarek committed:
Do not decode gpg output for processing

Process gpg output as binary data, to avoid decoding errors. Especially,
when the output includes binary NOTATION_DATA from sequoia.

marmarek committed:
Add vm-fc44 to example configs for R4.3 and main

QubesOS/qubes-issues#10703

marmarek committed:
ci: add cache job for fc44

QubesOS/qubes-issues#10703

marmarek committed:
Add Fedora 44 key and mock config

QubesOS/qubes-issues#10703

marmarek committed:
configs: switch stable kernel branch to 6.18

Do it in all three configs: 4.2, 4.3, main

QubesOS/qubes-issues#10713

unman committed:
Debian - add support for forky

fepitre committed:
pytest: make the tests verbose and add logs to artifacts

marmarek committed:
Merge remote-tracking branch 'origin/pr/228'

* origin/pr/228:
  pytest: make the tests verbose and add logs to artifacts
  chroot_archlinux: irgnore if an error occured on creating archive

fepitre committed:
test_executors: reformat

fepitre committed:
example-configs: update windows

marmarek committed:
Add keys for devel repo too

This allows setting:

    use-qubes-repo:
      version: devel
      testing: true

QubesOS/qubes-issues#10478

fepitre committed:
tests: remove 'quilt' from cache chroot of bookworm

marmarek committed:
Merge remote-tracking branch 'origin/pr/235'

* origin/pr/235:
  tests: remove 'quilt' from cache chroot of bookworm

fepitre committed:
tools: add verbose mode for windows scripts

marmarek committed:
Merge remote-tracking branch 'origin/pr/220'

* origin/pr/220:
  windows: change script line endings to LF
  windows: don't use redundant directories for build output
  windows: minor logging improvements
  windows: properly catch build errors
  Remove +x attribute from python file

Pull request description:

- (Finally) proper detection of msbuild errors
- remove unnecessary output copying during local builds
- misc cleanup

fepitre committed:
Add init-cache, sign and publish Windows plugins

This is to be compliant with other distribution workflows.
Add also specific tests to validate the function.
Stream logging and add helper scripts for tests.

We improve the SSHWindowsExecutor to support fully standalone
Windows machine. This machine needs to have OpenSSH configured
and user SSH key copied already.

Update the README for Windows.

Related to QubesOS/qubes-issues#10474

fepitre committed:
Add ansible support to setup the whole Qubes builder v2

fepitre committed:
executors/container: move image check/pull from __init__ to run()

Defer container image validation to runtime so executor can be instantiated
before image availability.

fepitre committed:
Add Windows CI job

marmarek committed:
Merge remote-tracking branch 'origin/pr/232'

* origin/pr/232:
  Debian - add support for forky

fepitre committed:
tests: update variables and versions

fepitre committed:
tests: switch CI tests to use example-advanced component

- Clean stale publish/repo state before template publish test.
- Update .gitlab-ci.yml: workflow rules, default values, fc40 -> fc43.

fepitre committed:
source_deb: support non-archive extra files with create-archive

When create-archive is true and files lists non-archive entries (e.g.
a GPG key), the prep stage failed because it assumed files[0] was
always the orig archive and rejected its extension.

Move create_archive computation before the extension check. When
create_archive is true, ext defaults to .gz and extra files are moved
into the source directory for debian/rules to use. When false, the
first file still replaces the orig archive like before.

fepitre committed:
tools/windows: fix VERBOSE variable initialization

fepitre committed:
README: use /usr/local/etc/qubes-rpc to avoid bind-dirs

fepitre committed:
README: add restorecon after installing qubes-rpc services

fepitre committed:
README: document Windows upload configuration

fepitre committed:
ansible: clarify Windows template prerequisite

fepitre committed:
ansible: remove task guard about legacy var name usage

fepitre committed:
ansible: add signed tag verification step

fepitre committed:
ci: add optional Ansible deployment test job

fepitre committed:
README: note that c:\build must not exist in Windows dispvm template

fepitre committed:
ansible: install builder-dvm RPC services in /usr/local/etc/qubes-rpc

fepitre committed:
ansible: accept a list of fingerprints for builder_git_verify_keys

marmarek committed:
Merge remote-tracking branch 'origin/pr/234'

* origin/pr/234:
  ansible: accept a list of fingerprints for builder_git_verify_keys
  ansible: install builder-dvm RPC services in /usr/local/etc/qubes-rpc
  README: note that c:\build must not exist in Windows dispvm template
  ci: add optional Ansible deployment test job
  ansible: add signed tag verification step
  ansible: remove task guard about legacy var name usage
  ansible: clarify Windows template prerequisite
  README: document Windows upload configuration
  README: add restorecon after installing qubes-rpc services
  README: use /usr/local/etc/qubes-rpc to avoid bind-dirs
  tools/windows: fix VERBOSE variable initialization
  tests: update variables and versions
  Add Windows CI job
  executors/container: move image check/pull from __init__ to run()
  Add ansible support to setup the whole Qubes builder v2
  Add init-cache, sign and publish Windows plugins
  tools: add verbose mode for windows scripts
  example-configs: update windows
  test_executors: reformat

fepitre committed:
plugin: introduce PluginContext and pipeline dependency graph

- jobs.py: introduces Pipeline (jobs container à la Gitlab), JobFactory (plugin instantiation), and cycle detection that replace the heavy dummy logic that was done config.py.
- DAG: topological sort and dependency resolution moved into Pipeline.sorted_jobs() / build_graph().
- New pipeline CLI subcommand that allow to show what would run without executing anything.
- Remove unused imports.

Fixes QubesOS/qubes-issues#10522
Fixes QubesOS/qubes-issues#10555

fepitre committed:
ci: trigger builderv2-github on deploy stage

marmarek committed:
Merge remote-tracking branch 'origin/pr/238'

* origin/pr/238:
  source_deb: support non-archive extra files with create-archive
  tests: switch CI tests to use example-advanced component

fepitre committed:
ci: start cache jobs after pytest-cli-component jobs

Cache jobs wait for pytest-cli-component* jobs when they exist in the
pipeline. optional: true so cache jobs still work when pytest is
disabled via CI_RUN_PYTEST_JOBS=0.

fepitre committed:
ci: start template jobs after pytest-cli-template jobs

Template jobs wait for pytest-cli-template* jobs when they exist in the
pipeline. optional: true so template jobs still work when pytest is
disabled via CI_RUN_PYTEST_JOBS=0.

fepitre committed:
ci: use CI_PROJECT_DIR instead of hardcoded runner path in local job

fepitre committed:
fetch: fix ambiguous - placement in component_directory regex

fepitre committed:
jobs: sort jobs by stage and component declaration order

Replace TopologicalSorter with Kahn's algorithm using a priority queue.
Jobs are picked in stage order; within the same stage, components run in
the order they are declared in the config. Implicit stage edges are added
so later stages always run after earlier ones for the same target, even
when an intermediate stage is absent.

fepitre committed:
publish_windows: skip when no repository is configured

marmarek committed:
Merge remote-tracking branch 'origin/pr/237'

* origin/pr/237:
  publish_windows: skip when no repository is configured
  jobs: sort jobs by stage and component declaration order
  fetch: fix ambiguous - placement in component_directory regex
  ci: use CI_PROJECT_DIR instead of hardcoded runner path in local job
  ci: start template jobs after pytest-cli-template jobs
  ci: start cache jobs after pytest-cli-component jobs
  ci: trigger builderv2-github on deploy stage
  plugin: introduce PluginContext and pipeline dependency graph

Pull request description:

- jobs.py: introduces Pipeline (jobs container à la Gitlab), JobFactory (plugin instantiation), and cycle detection that replace the heavy dummy logic that was done config.py.
- DAG: topological sort and dependency resolution moved into Pipeline.sorted_jobs() / build_graph().
- New pipeline CLI subcommand that allow to show what would run without executing anything.

Fixes QubesOS/qubes-issues#10522
Fixes QubesOS/qubes-issues#10555

marmarek committed:
Restore documented behavior of skip-git-fetch

The supposed behavior is:

    When set, do not update already downloaded git repositories (those
    in `sources` artifacts dir). New components are still fetched
    (once).

Do not skip initial git fetch when the source is not downloaded yet.

This reverts small part of 903b7bf "plugin: introduce PluginContext and
pipeline dependency graph".

marmarek committed:
Add key for qubes-devel for ubuntu too

QubesOS/qubes-issues#10478

marmarek committed:
Update qubes-infrastructure-mirrors submodule

marmarek committed:
Update qubes-infrastructure-mirrors again

fepitre committed:
executors: fix resource leak in container client context manager

fepitre committed:
cli: fix return-code handling for single-command invocations

fepitre committed:
chroot: add install-packages flag to cache configuration

fepitre committed:
list-deps: introduce base plugin and stage framework

fepitre committed:
list-deps: add RPM, DEB and Arch Linux dist plugins

fepitre committed:
list-deps: expose as a CLI command

fepitre committed:
README: add list-deps CLI and options to install pkgs

fepitre committed:
Make mypy happy

fepitre committed:
tests: cover list-deps plugin and CLI

Harden also dependencies validations:
- reject parens, path traversal (..), and overlong names
- quote packages in apt-download-packages shell command

fepitre committed:
ansible: add support for docker executor setup

marmarek committed:
Add devel repo definition/key for Archlinux too

Arch has repository name embedded into repository itself (in .db file
name), so it needs to match in the config - "rdevel" is not good enough.

QubesOS/qubes-issues#10478

fepitre committed:
config: support empty-dict override via -o key= or -o key={} in CLI options

fepitre committed:
chroot_rpm: save pre-installed deps as root_cache_install to avoid overwriting minimal root_cache

fepitre committed:
chroot_deb/build_deb: bake qubes keyring into base.tgz and skip pbuilder update when install-packages is set

marmarek committed:
Fix generating pacman.conf

Don't parse use_qubes_repo_version when not set.

Fixes: d7b0b3c "Add devel repo definition/key for Archlinux too"

fepitre committed:
ansible: install archlinux devtools on the executor template

Mirrors the qubes-builder fedora docker image: clones devtools at a
specified commit, runs 'make install', and symlinks archbuild as
qubes-x86_64-build.

fepitre committed:
ci: add benchmark script for builds with/without cache

fepitre committed:
Make python-black happy

fepitre committed:
list-deps archlinux: use only build deps, drop runtime depends

fepitre committed:
list-deps rpm: drop '|| true' on generate-spec

fepitre committed:
list-deps: add --exclude option, default to dropping qubes-* packages

fepitre committed:
README: document list-deps --exclude option and per-dist caveats

Cover the qubes-* default exclude, the --exclude / list-deps.exclude
override path, the DEB version-stripping limitation, and the RPM
virtual-provide drop.

marmarek committed:
Merge remote-tracking branch 'origin/pr/240'

* origin/pr/240:
  ansible: install archlinux devtools on the executor template
  ansible: add support for docker executor setup

fepitre committed:
ansible: fix windows_builder_dvm_name in RPC policy templates

fepitre committed:
tests: add more list-deps integration tests

fepitre committed:
publish_deb: clean stale devel sources across the shared pool

orig.tar.gz has no devel suffix, so any other dist sharing the
reprepro pool keeps pinning the previous one and reprepro rejects the
new orig with conflicting checksums.

Find every Debian/Ubuntu dist sharing the pool and drop the stale
source so the orig.tar.gz is freed. Binary .debs stay published.

QubesOS/qubes-issues#10555

marmarek committed:
Merge remote-tracking branch 'origin/pr/241'

* origin/pr/241:
  ansible: fix windows_builder_dvm_name in RPC policy templates

marmarek committed:
example-configs: (re-)add pykickstart package

Update pykickstart in R4.3+, to include a few fixes.

QubesOS/qubes-issues#8993
QubesOS/qubes-issues#10516

marmarek committed:
Merge remote-tracking branch 'origin/pr/242'

* origin/pr/242:
  publish_deb: clean stale devel sources across the shared pool

fepitre committed:
tests: add chroot_{rpm,deb,archlinux} install-packages init-cache test

fepitre committed:
tests: cover list_deps skip and no-packages branches

fepitre committed:
tests: cover container executor unknown-client and get_user/get_group

fepitre committed:
tests: cover cli_main option-parser branches

fepitre committed:
reproduce DB_NOTFOUND files from deb.qubes-os.org/devel

Here are the commands:
mkdir -p tests/fixtures/devel-vm/pool/main/q/qubes-utils
rsync rsync://ftp.qubes-os.org/qubes-mirror/repo/deb/devel/vm/pool/main/q/qubes-utils/qubes-utils_4.3.16+deb12u1+devel{8,9}.{dsc,tar.xz} tests/fixtures/devel-vm/pool/main/q/qubes-utils/

fepitre committed:
publish_deb: tolerate removetrack failure on stale tracking entry

fepitre committed:
tests: cover is_safe_dep path traversal and update non-dict cache

marmarek committed:
Merge remote-tracking branch 'origin/pr/239'

* origin/pr/239: (24 commits)
  tests: cover is_safe_dep path traversal and update non-dict cache
  tests: cover cli_main option-parser branches
  tests: cover container executor unknown-client and get_user/get_group
  tests: cover list_deps skip and no-packages branches
  tests: add chroot_{rpm,deb,archlinux} install-packages init-cache test
  tests: add more list-deps integration tests
  README: document list-deps --exclude option and per-dist caveats
  list-deps: add --exclude option, default to dropping qubes-* packages
  list-deps rpm: drop '|| true' on generate-spec
  list-deps archlinux: use only build deps, drop runtime depends
  Make python-black happy
  ci: add benchmark script for builds with/without cache
  chroot_deb/build_deb: bake qubes keyring into base.tgz and skip pbuilder update when install-packages is set
  chroot_rpm: save pre-installed deps as root_cache_install to avoid overwriting minimal root_cache
  config: support empty-dict override via -o key= or -o key={} in CLI options
  tests: cover list-deps plugin and CLI
  Make mypy happy
  README: add list-deps CLI and options to install pkgs
  list-deps: expose as a CLI command
  list-deps: add RPM, DEB and Arch Linux dist plugins
  ...

Pull request description:

Related to:
  - https://github.com/QubesOS/qubes-issues/issues/9702
  - https://github.com/QubesOS/qubes-issues/issues/9703

marmarek committed:
Refresh mirrorlist

This is used to generate links in metalink file.

marmarek committed:
tests: configure git signing only once

Not for each commit over and over.

marmarek committed:
tests: move home_directory fixture to conftest

Will be useful in other tests

marmarek committed:
tests: don't look for conftest.py in artifacts

It takes a while if it's fully populated, so avoid the search.

marmarek committed:
tests: check if git archive is correctly created

Check if .gitignore is correctly parsed, including re-including files.

marmarek committed:
fetch: pack all of the git content

Linux 7.1 includes negated entries in .gitignore, which tar
--exclude-vcs-ignores doesn't support. Since it's literally fresh clone,
the simplest option is to just don't use .gitignore, as there should be
no extra files.

marmarek committed:
Merge remote-tracking branch 'origin/pr/243'

* origin/pr/243:
  reproduce DB_NOTFOUND files from deb.qubes-os.org/devel
  publish_deb: tolerate removetrack failure on stale tracking entry

marmarek committed:
iso: make detached signature on digests file, instead of cleartext one

QubesOS/qubes-issues#10512

fepitre committed:
cleanup: prune outdated installer chroot and templates cache

Add three options to CLI cleanup cache:

  --installer-chroot-only-unused     drop chroots for dists not in
                                     the current builder.yml host set
  --installer-templates-only-unused  drop cached templates whose name is
                                     no longer in cache.templates
  --installer-templates-only-old     keep only the newest RPM per
                                     template name in the cache

`qb cleanup all` now runs the three prunes plus --installer-bootstrap
by default. Use --no-XXX to restore the previous behavior.

Fixes QubesOS/qubes-issues#10878

marmarek committed:
iso: drop weak hashes from digests file

marmarek committed:
Merge remote-tracking branch 'origin/pr/248'

* origin/pr/248:
  cleanup: prune outdated installer chroot and templates cache

marmarek committed:
Merge branch 'digests'

* digests:
  iso: drop weak hashes from digests file
  iso: make detached signature on digests file, instead of cleartext one

marmarek committed:
Merge remote-tracking branch 'origin/pr/193' into torrent

* origin/pr/193:
  installer: rewrite create-torrent

Pull request description:

This is Draft.
Current create-torrent problems:
- On Debian, creating v2 torrent isn't supported.(And mktorrent is outdated software)
- Setting custom webseed hosts isn't supported.

Creating Python script which uses libtorrent-rasterbar Python bindings will resolve both.
- Both Debian(12+) and Fedora provides libtorrent-rasterbar python bindings package
- Setting custom properties is easier

This is draft because supporting custom webseed url is in progress.
Any reviews are welcome!

marmarek committed:
Fixups to the create-torrent script

Fix config usage, and create_torrent_qubes function call.

marmarek commented:
Ouch, github got confused about the merge commit... Anyway, I resolved conflict and fixed some issues with the script. Now it appears to work.

I also tried to include the  `DIGESTS.asc` file, but then transmission can't open the file (not much details, just "Couldn't add torrent file '/home/user/Downloads/Qubes-202606100121-x86_64.torrent'"). But `gtorrentviewer` sees it correctly. Idk what happens there, but I'm not going to investigate too much - detached signature of the ISO file itself is included already, so it isn't a huge deal.

Another thing I see, is a `.pad` directory, that Transmission wants to download. Is it something new in this format that Transmission specifically don't know how to handle, or a bug in the script?

zpc0 commented:
It seems that Transmission have `.pad` dir problem...
https://github.com/transmission/transmission/issues/4602

Could you try `qBittorrent`?

marmarek commented:
Well, it's not about what _I_ can try, we want a torrrent file that works with (at least) most popular clients...

zpc0 commented:
> Well, it's not about what _I_ can try, we want a torrrent file that works with (at least) most popular clients...

Indeed.
We need to wait for more v2 support on client side...
